src/irr/core/LRUcache.h

```diff
@@ -0,0 +1,131 @@
+#ifndef __LRU_CACHE_H_INCLUDED__
+#define __LRU_CACHE_H_INCLUDED__
+
+#include "irr/core/CFixedCapacityDoublyLinkedList.h"
+
+namespace irr {
+	namespace core {
+
+//Key-Value LRU cache
+//Stores fixed size amount of elements. 
+//When inserting new, prevents overflow by removing the least recently used element.
+template<typename Key, typename Value, typename MapHash = std::hash<Key>, typename MapEquals = std::equal_to<Key>>
+class LRUcache
+{
+	typedef std::pair<Key, Value> list_template_t;
+	_IRR_STATIC_INLINE_CONSTEXPR uint32_t invalid_iterator = Snode<list_template_t>::invalid_iterator;
+
+	DoublyLinkedList<list_template_t> m_list;
+	unordered_map<Key, uint32_t, MapHash, MapEquals> m_shortcut_map;
+	uint32_t m_capacity;
+
+	//get iterator associated with a key, or invalid_iterator if key is not within the cache
+	inline uint32_t common_peek(Key& key)
+	{
+		if (m_shortcut_map.find(key) != m_shortcut_map.end())
+		{
+			return m_shortcut_map[key];
```

Comment:
return the dereferenced found iterator, this `operator[]` will trigger yet another search that you don't need.